main.py

```diff
@@ -35,26 +42,36 @@
 # [START run_pubsub_handler]
 @app.route('/', methods=['POST'])
 def index():
-    envelope = request.get_json()
-    if not envelope:
-        msg = 'no Pub/Sub message received'
-        print(f'error: {msg}')
-        return f'Bad Request: {msg}', 400
+    pubsub_received_message = request.get_json()
+
+    # parse the Pub/Sub data
+    try:
+        pubsub_data_string = pubsub.parse_data_from_message(pubsub_received_message)
+    except pubsub.DataParseError as e:
+        print(e)
+        return (e.message, 400)
+
+    # load the notification from the data
+    try:
+        monitoring_notification_dict = json.loads(pubsub_data_string)
+    except json.JSONDecodeError as e:
+        msg = 'notification could not be decoded to a JSON'
+        print(msg)
```

Comment:
Yeah I was a bit conflicted on how to handle this, because I definitely remembered your suggestion from before. I think I wanted to be able to include a more notification-specific message for the error, but I didn't really know how to do that unless I raised a custom exception, which I could do but feels a little unnecessary since it's so specific. Being able to check that my message is printed in the test cases was also useful. The stacktrace is useful though so I'm thinking one way to fix this is to print(e) but return (msg, 400) and in the test cases instead of checking the stdout output, I could just check the message in the returned response? This way I can still use a custom message but the stacktrace is still printed in stdout.